packages/contracts-bedrock/scripts/deploy/DeployImplementations.s.sol

```diff
@@ -418,6 +434,8 @@ contract DeployImplementations is Script {
 
     function run(DeployImplementationsInput _dii, DeployImplementationsOutput _dio) public {
         // Deploy the implementations.
+        deploySuperchainConfigImpl(_dio);
+        deployProtocolVersionsImpl(_dio);
```

Comment:
It feels like a smell that we have these superchain impl deployments here and in `DeploySuperchain.s.sol`, though I see why you did it this way.

We do deployments through these public, modular `deploy*` methods so they can be easily reused and composed as needed needed. So instead of re-implementing these methods here we should just import and use `deploySuperchainImplementationContracts` (or, split that up into `deploySuperchainConfigImpl` and `deployProtocolVersionsImpl` and then import those two methods)

Reply:
I had done something very much like that. Here's [the commit](https://github.com/ethereum-optimism/optimism/commit/fde734cdbe7972011e9b75607e6fc0c13b4c64f5#diff-9faeded4d5391028a1ec20dd3c460f0547354f179c585370ed07816e0dd74ebdL344) where I undid it.

To me it felt a bit weird, in the sense that, yes there is some duplication, but these scripts already have a lot of boilerplate and so having to jump to another file to read the logic didn't feel like that a net benefit.